res/controllers/hid-controller-api.d.ts

```diff
@@ -0,0 +1,47 @@
+
+declare class HidControllerJSProxy {
+    /** Sends HID OutputReport with hard coded ReportID 0 to HID device
+     *        This function only works with HID devices, which don't use ReportIDs
+     * @param dataList Data to send as list of bytes
+     * @param length This optional argument is no longer evaluated, and only here for backwards compatibility
+    */
+    public send(dataList:number[], length:number=0):void;
+
+    /** Sends HID OutputReport to HID device
+     * @param dataList Data to send as list of bytes
+     * @param length Unused but mandatory argument
+     * @param reportID 1...255 for HID devices that uses ReportIDs - or 0 for devices, which don't use ReportIDs
+     * @param resendUnchangedReport If set, the report will also be send, if the data are unchanged since last sending
+    */
+    public send(dataList:number[], length:number, reportID:number, resendUnchangedReport:boolean = false):void;
+
+    /** Sends an OutputReport to HID device
+     *  @param reportID 1...255 for HID devices that uses ReportIDs - or 0 for devices, which don't use ReportIDs
+     *  @param dataArray Data to send as byte array (Javascript type Uint8Array)
+     *  @param resendUnchangedReport If set, the report will also be send, if the data are unchanged since last sending
+     */
+    public sendOutputReport(reportID:number, dataArray:ArrayBuffer, resendUnchangedReport:boolean = false):void;
+
+    /** getInputReport receives an InputReport from the HID device on request.
+     *  @details This can be used on startup to initialize the knob positions in Mixxx
+     *           to the physical position of the hardware knobs on the controller.
+     *           This is an optional command in the HID standard - not all devices support it.
+     *  @param reportID 1...255 for HID devices that uses ReportIDs - or 0 for devices, which don't use
+     *  @return Returns report data with ReportID byte as prefix
+     */
+    public getInputReport(reportID:number):Uint8Array;
+
+    /** Sends a FeatureReport to HID device
+     *  @param reportID 1...255 for HID devices that uses ReportIDs - or 0 for devices, which don't use
+     *  @param reportData Data to send as byte array
+     */
+    public sendFeatureReport(reportID:number, reportData:ArrayBuffer):void;
+
+    /** getFeatureReport receives a FeatureReport from the HID device on request.
+     *  @param reportID 1...255 for HID devices that uses ReportIDs - or 0 for devices, which don't use
```

Comment:
This seems not to be possible, I tried various variants and none worked

Reply:
Can you elaborate? Doesn't `type reportID = number` work?

Reply:
Nope, I tried this in several variants. It just documents the type, the documentation is not taken over to the argument: 
![grafik](https://user-images.githubusercontent.com/64457745/178159494-1cce9ef2-4326-4956-9013-6acffa8b1759.png)

Reply:
Only where the name of the type is used in the sourcecode itself, I get the description. This has no worth for a mapping developer. He needs an explanation, how to use the API - a documentation of all function arguments.

Reply:
Right... but if a type is used everywhere equally, we should only have to document it once. Not the same doc each time its used. The question is: is there a way to present this doc to the developer within the editor without making it the description of the function argument?

Reply:
I spend a lot of time today, to find such a solution - I did not find any. I don't think, the language has such a feature.
I will continue with the other review remarks instead.

Reply:
> I spend a lot of time today, to find such a solution - I did not find any. I don't think, the language has such a feature.

Yes, its not really a language feature, its an editor feature, since that's whats you are optimizing for at the moment. Still I don't believe copy pasting the same description 10 times makes sense...

Reply:
Rethinking this I came to the conclusion to relax my DRY-ness obsession for this. The primary objective of this PR is to improve the developer experience and thus we should optimize for that. The documented APIs have to be stable anyways so its not like their docs need to be updated often anyways.

Reply:
Rethinking this I came to the conclusion to relax my DRY-ness obsession for this. The primary objective of this PR is to improve the developer experience and thus we should optimize for that. The documented APIs have to be stable anyways so its not like their docs need to be updated often anyways.